sim/dex-data.ts

```diff
@@ -141,8 +141,8 @@ export class BasicEffect implements EffectData {
 		this.duration = data.duration;
 		this.noCopy = !!data.noCopy;
 		this.affectsFainted = !!data.affectsFainted;
-		this.status = data.status || undefined;
-		this.weather = data.weather || undefined;
+		this.status = toId(data.status) || undefined;
```

Comment:
I'm not a huge fan of unnecessary `toId` calls; Dex constructors are an inner loop.

Reply:
Yeah, my problem was `EffectData` can't be given the `ID` type (without `/** @type {ID} */(...)` casts EVERYWHERE, or `as ID` when Typescript-ed).

However, maybe we can just cast here instead of `toId` and just assume (or add a test to validate) that the data files are in fact correct IDs?

Reply:
Changes this to a cast